lib/web/apiserver.go

```diff
@@ -2142,8 +2142,56 @@ func (h *Handler) siteNodeConnect(
 		return nil, trace.Wrap(err)
 	}
 
+	// Fetching nodes is quite slow when many nodes exist so fetch it once and
+	// pass the value to the methods that require it.
+	clt, err := ctx.GetUserClient(site)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+
+	namespace := apidefaults.Namespace
+	if req.Namespace != "" {
```

Comment:
We don't support namespaces, so you can always set this to default.

Reply:
I've removed Namespace from the `TerminalRequest` type and set it to the default.